Unable to build, "multiple definition of rust_eh_personality"

### Summary

I haven't been able to build Ladybird successfully anymore as of late. On a fresh clone of the master branch (commit c701781d1194a38d995e1ec36c557846025e08bd), I'm hitting this error:

```
$ ./Meta/ladybird.py build --jobs=16
ninja: Entering directory `/home/user/Projects/ladybird/Build/release'
[0/2] Re-checking globbed directories...
[1/74] Linking CXX shared library lib64/liblagom-web.so.0.1.0
FAILED: [code=1] lib64/liblagom-web.so.0.1.0 
: && /usr/bin/c++ -fPIC -O2 -g -DNDEBUG  -shared -fstack-protector-strong -Wl,-z,defs -Wl,--no-undefined -Wl,--no-allow-shlib-undefined -Wl,--dependency-file=Libraries/LibWeb/CMakeFiles/LibWeb.dir/link.d -Wl,-soname,liblagom-web.so.0 -o lib64/liblagom-web.so.0.1.0 @CMakeFiles/LibWeb.rsp && :
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o): in function `rust_eh_personality':
libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0:(.text.rust_eh_personality+0x0): multiple definition of `rust_eh_personality'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/personality/gcc.rs:299: first defined here
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/panicking.rs:43: multiple definition of `std::panicking::EMPTY_PANIC'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/panicking.rs:43: first defined here
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/args/unix.rs:121: multiple definition of `std::sys::args::unix::imp::ARGV_INIT_ARRAY'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/args/unix.rs:121: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Operating system

Linux

### Steps to reproduce

1. Clone Ladybird
2. Build ladybird using 'ladybird.py build'
3. See the error


### Expected behavior

Ladybird should build successfully

### Actual behavior

Ladybird fails to build with the error specified earlier.

### URL for a reduced test case

N/A

### HTML/SVG/etc. source for a reduced test case

```html
N/A
```

### Log output and (if possible) backtrace


```shell
Stdout when building:

ninja: Entering directory `/home/user/Projects/ladybird/Build/release'
[0/2] Re-checking globbed directories...
[1/74] Linking CXX shared library lib64/liblagom-web.so.0.1.0
FAILED: [code=1] lib64/liblagom-web.so.0.1.0 
: && /usr/bin/c++ -fPIC -O2 -g -DNDEBUG  -shared -fstack-protector-strong -Wl,-z,defs -Wl,--no-undefined -Wl,--no-allow-shlib-undefined -Wl,--dependency-file=Libraries/LibWeb/CMakeFiles/LibWeb.dir/link.d -Wl,-soname,liblagom-web.so.0 -o lib64/liblagom-web.so.0.1.0 @CMakeFiles/LibWeb.rsp && :
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o): in function `rust_eh_personality':
libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0:(.text.rust_eh_personality+0x0): multiple definition of `rust_eh_personality'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/personality/gcc.rs:299: first defined here
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/panicking.rs:43: multiple definition of `std::panicking::EMPTY_PANIC'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/panicking.rs:43: first defined here
/usr/bin/ld.bfd: cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_content_blocker_rust.a(libweb_content_blocker_rust-f18bafd6e92a6a19.libweb_content_blocker_rust.a9cb22dd0f765afa-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/args/unix.rs:121: multiple definition of `std::sys::args::unix::imp::ARGV_INIT_ARRAY'; cargo/build/x86_64-unknown-linux-gnu/release/liblibweb_rust.a(libweb_rust-fd82b052b9ab7380.libweb_rust.c446f7da9e7c4bcb-cgu.0.rcgu.o):/builddir/build/BUILD/rust-1.95.0-build/rustc-1.95.0-src/library/std/src/sys/args/unix.rs:121: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Screenshots or screen recordings

_No response_

### Build flags or config settings

/etc/profile.d/env.sh
```
export CARGO_HOME=$HOME/.local/share/cargo
export HISTFILE=$HOME/.local/share/bash/bash_history
export XDG_CACHE_HOME=$HOME/.local/share/cache
export XDG_CONFIG_HOME=$HOME/.config
export XDG_DATA_HOME=$HOME/.local/share
export XDG_STATE_HOME=$HOME/.local/state
```

$HOME/.local/share/cargo/config.toml
```
[profile.dev]
debug = false

[profile.release]
codegen-units = 1
lto = true
panic = "abort"
strip = true
```

[fedora-packages.txt](https://github.com/user-attachments/files/28258126/fedora-packages.txt)

### Contribute a patch?

- [ ] I’ll contribute a patch for this myself.

--- a/Meta/CMake/rust_crate.cmake
+++ b/Meta/CMake/rust_crate.cmake
@@ -54,6 +54,7 @@ function(import_rust_crate)
                 -P "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/sync_rust_ffi_header.cmake"
         DEPENDS "${manifest_path}"
             "${workspace_dir}/Cargo.lock" "${workspace_dir}/Cargo.toml"
+            "${RUST_RUSTC}" "${CMAKE_SOURCE_DIR}/rust-toolchain.toml"
         DEPFILE "${depfile}"
         COMMENT "Building Rust crate ${ARG_CRATE_NAME}"
         USES_TERMINAL
@@ -108,6 +109,7 @@ function(build_rust_binary)
         COMMAND ${CMAKE_COMMAND} -E copy_if_different "${cargo_binary}" "${output_binary}"
         DEPENDS "${manifest_path}"
             "${workspace_dir}/Cargo.lock" "${workspace_dir}/Cargo.toml"
+            "${RUST_RUSTC}" "${CMAKE_SOURCE_DIR}/rust-toolchain.toml"
         DEPFILE "${depfile}"
         COMMENT "Building Rust binary ${ARG_BINARY_NAME}"
         USES_TERMINAL

